src/include/sof/audio/mux.h

```diff
@@ -55,7 +55,8 @@ struct mux_stream_data {
 	uint8_t num_channels_deprecated;	/* deprecated in ABI 3.15 */
 	uint8_t mask[PLATFORM_MAX_CHANNELS];
 
-	uint8_t reserved[(20 - PLATFORM_MAX_CHANNELS - 1) % 4]; // padding to ensure proper alignment of following instances
+	uint8_t reserved1[8 - PLATFORM_MAX_CHANNELS]; // padding for extra channels
+	uint8_t reserved2[3]; // padding to ensure proper alignment of following instances
 };
```

Comment:
A general question: why does this structure have reserved fields? Is it a part of an ABI? If it is, usually such structures use `__attribute__((packed))`, so it might be desirable to add one here.

Reply:
I agree __attribute__((packed)) is missing but did not add it due to it being absent throughout the file.

Reply:
The comment on the above field suggests that it is, so that is concerning.

Reply:
@afq984 can we fix the missing attribute. Thanks.

Reply:
Done